install.sh

```diff
@@ -1,41 +1,57 @@
 #!/usr/bin/env bash
+
 set -e
 
-resolve_link() {
-  $(type -p greadlink readlink | head -n1) "$1"
+PREFIX="$1"
+if [ -z "$1" ]; then
+  { echo "usage: $0 <prefix>"
+    echo "  e.g. $0 /usr/local"
+  } >&2
+  exit 1
+fi
+
+export BATS_READLINK='true'
+if command -v 'greadlink' >/dev/null; then
+  BATS_READLINK='greadlink'
+elif command -v 'readlink' >/dev/null; then
+  BATS_READLINK='readlink'
+fi
+
+bats_resolve_link() {
+  if ! "$BATS_READLINK" "$1"; then
+    return 0
+  fi
 }
 
-abs_dirname() {
-  local cwd="$(pwd)"
+bats_absolute_path() {
+  local cwd="$PWD"
   local path="$1"
 
-  while [ -n "$path" ]; do
+  while [[ -n "$path" ]]; do
     cd "${path%/*}"
     local name="${path##*/}"
-    path="$(resolve_link "$name" || true)"
+    path="$(bats_resolve_link "$name")"
   done
 
-  pwd
+  printf -v "$2" -- '%s' "$PWD"
   cd "$cwd"
 }
 
-PREFIX="$1"
-if [ -z "$1" ]; then
-  { echo "usage: $0 <prefix>"
-    echo "  e.g. $0 /usr/local"
-  } >&2
-  exit 1
-fi
+bats_install_scripts() {
+  local scripts_dir
+  local scripts=()
+
+  for scripts_dir in "$@"; do
+    scripts=("$BATS_ROOT/$scripts_dir"/*)
+    cp "${scripts[@]}" "$PREFIX/$scripts_dir"
+    chmod a+x "${scripts[@]/#$BATS_ROOT[/]/$PREFIX/}"
+  done
+}
 
-BATS_ROOT="$(abs_dirname "$0")"
 mkdir -p "$PREFIX"/{bin,libexec,share/man/man{1,7}}
-cp -R "$BATS_ROOT"/bin/* "$PREFIX"/bin
-cp -R "$BATS_ROOT"/libexec/* "$PREFIX"/libexec
-cp "$BATS_ROOT"/man/bats.1 "$PREFIX"/share/man/man1
-cp "$BATS_ROOT"/man/bats.7 "$PREFIX"/share/man/man7
-
-# fix file permission
-chmod a+x "$PREFIX"/bin/*
-chmod a+x "$PREFIX"/libexec/*
+bats_absolute_path "$0" 'BATS_ROOT'
```

Comment:
I'm not following how `bats_absolute_path` works to set BATS_ROOT. (And my macOS/bsd `printf` doesn't have `-v` documented... I think these things are related 😄 )

Reply:
Ah, that's because `printf` is a Bash builtin, and the `-v` option takes the name of a variable into which to store the result. This is much quicker than capturing stdout into a variable via `$()`.

Was that the only question you had about `bats_absolute_path`? Happy to unpack it some more.

Reply:
Thanks! That was all I didn't quite grok. And got lazy trying to find docs on `-v` (even a cursory google search didn't include -v in their docs, so thanks for the lazyweb explain 😄 )